### PR TITLE
EVA-3291 - Use job tracker instead of logs to get job timestamps

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**

### CODE REVIEW

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 1]
```diff
@@ -69,81 +99,146 @@ def get_assembly_info_and_date_ranges(all_log_files):
...
+                    [result["job_start_time"] for result in all_results
+                     if result["job_id"] == job_id and result["job_start_time"]] +
+                    [result["step_end_time"] for result in all_results
+                     if result["job_id"] == job_id and result["step_end_time"]] +
+                    [result["step_start_time"] for result in all_results
+                     if result["job_id"] == job_id and result["step_start_time"]])
+        # Fill in start/end times for clustering jobs
+        for job_info in set([(result["job_id"], result["job_name"], result["job_start_time"], result["job_end_time"])
+                             for result in all_results]):
+            time_ranges_by_job_id[job_info[0]][job_info[1]] = {"launched": job_info[2]}
```
Comment (sundarvenkata-EBI) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:229]: See [here](https://github.com/EBIvariation/eva-accession/blob/b73d13edda9c305ee2d9dbf5a860793acd524212/eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py#L436) for the previous approach

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 2]
```diff
@@ -69,81 +99,146 @@ def get_assembly_info_and_date_ranges(all_log_files):
...
+            # Fill out completed time only if it is not null
+            if job_info[3] is not None:
+                time_ranges_by_job_id[job_info[0]][job_info[1]]["completed"] = job_info[3]
+        # Fill in start/end times for clustering steps
+        for step_info in set([(result["job_id"], result["step_name"], result["step_start_time"])
+                              for result in all_results]):
+            time_ranges_by_job_id[step_info[0]][step_info[1]] = step_info[2]
```
Comment (sundarvenkata-EBI) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:236]: See [here](https://github.com/EBIvariation/eva-accession/blob/b73d13edda9c305ee2d9dbf5a860793acd524212/eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py#L445) for the previous approach.

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 3]
```diff
@@ -69,81 +99,146 @@ def get_assembly_info_and_date_ranges(all_log_files):
...
+            time_ranges_by_job_id[job_id] = defaultdict(dict)
+            time_ranges_by_job_id[job_id]["last_timestamp"] = \
+                max([result["job_end_time"] for result in all_results
+                     if result["job_id"] == job_id and result["job_end_time"]] +
+                    [result["job_start_time"] for result in all_results
+                     if result["job_id"] == job_id and result["job_start_time"]] +
+                    [result["step_end_time"] for result in all_results
+                     if result["job_id"] == job_id and result["step_end_time"]] +
+                    [result["step_start_time"] for result in all_results
+                     if result["job_id"] == job_id and result["step_start_time"]])
```
Comment (sundarvenkata-EBI) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:225]: See [here](https://github.com/EBIvariation/eva-accession/blob/b73d13edda9c305ee2d9dbf5a860793acd524212/eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py#L451) for the previous approach.

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 4]
```diff
@@ -1,39 +1,69 @@
...
-    # Beginning release 4, clustering of new studies is also carried out during remapping
-    all_remapping_logs_pattern = os.path.join(remapping_dir, '*', 'GCA_*', 'logs', '*_clustering.log')
-    all_log_files = glob.glob(all_clustering_logs_pattern) + glob.glob(all_remapping_logs_pattern)
+def get_release_end_timestamps_from_last_release(private_config_xml_file, current_release_version) -> dict:
+    """
+    Get release end timestamps from last release for all assemblies
+    """
+    assembly_timestamp_map = {}
+    with psycopg2.connect(get_pg_metadata_uri_for_eva_profile("production_processing", private_config_xml_file),
+                          user="evapro") as metadata_connection_handle:
```
Comment (apriltuesday) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:29]:
```suggestion
    with get_metadata_connection_handle("production_processing", private_config_xml_file) as metadata_connection_handle:
```
and similar elsewhere

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 5]
```diff
@@ -1,39 +1,69 @@
...
+                                           "group by assembly_accession"
+        for result in get_all_results_for_query(metadata_connection_handle, prev_release_end_timestamp_query):
+            assembly_timestamp_map[result[0]] = result[1]
+    return assembly_timestamp_map
+
+
+def get_all_assemblies_for_current_release(private_config_xml_file, current_release_version):
+    with psycopg2.connect(get_pg_metadata_uri_for_eva_profile("production_processing", private_config_xml_file),
+                          user="evapro") as metadata_connection_handle:
+        query_to_get_release_end_timestamp = "select distinct assembly_accession from " \
```
Comment (apriltuesday) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:45]:
```suggestion
        query_to_get_assemblies_for_release = "select distinct assembly_accession from " \
```

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 6]
```diff
@@ -69,81 +99,146 @@ def get_assembly_info_and_date_ranges(all_log_files):
...
             }
     return ranges_per_assembly
 
 
+def get_time_ranges_by_job_id_from_job_tracker(assembly_accession, private_config_xml_file,
+                                               prev_release_end_timestamp_for_assembly: datetime = None):
+    # Job and step start/end times by job ID
+    time_ranges_by_job_id = defaultdict(dict)
+    url, username, password = get_accession_pg_creds_for_profile("production_processing", private_config_xml_file)
+    accessioning_job_tracker_handle = psycopg2.connect(urlsplit(url).path, user=username, password=password,
```
Comment (apriltuesday) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:181]: Should put in a `with` statement, or just make sure we close the connection at the end

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 7]
```diff
@@ -69,81 +99,146 @@ def get_assembly_info_and_date_ranges(all_log_files):
...
+                    [result["step_start_time"] for result in all_results
+                     if result["job_id"] == job_id and result["step_start_time"]])
+        # Fill in start/end times for clustering jobs
+        for job_info in set([(result["job_id"], result["job_name"], result["job_start_time"], result["job_end_time"])
+                             for result in all_results]):
+            time_ranges_by_job_id[job_info[0]][job_info[1]] = {"launched": job_info[2]}
+            # Fill out completed time only if it is not null
+            if job_info[3] is not None:
+                time_ranges_by_job_id[job_info[0]][job_info[1]]["completed"] = job_info[3]
+        # Fill in start/end times for clustering steps
```
Comment (apriltuesday) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:233]:
```suggestion
        # Fill in start times for clustering steps
```

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py [hunk 8]
```diff
@@ -292,9 +390,9 @@ def fill_data_for_current_release(metadata_connection_handle, metrics_per_assemb
 
         # Finally perform inserts, once for each taxonomy but otherwise identical
         for taxid, scientific_name in ranges_per_assembly[asm]['species_info']:
-            folder = f"{scientific_name}/{asm}"
-            formatted_name = scientific_name.capitalize().replace('_', ' ')
-            insert_query = f"insert into {assembly_table_name} "\
+            folder = f"{scientific_name}/{asm}".replace("'", "''")
```
Comment (apriltuesday) [inline eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:393]: Just curious, why are there sometimes quotes in the scientific name?

**Review by apriltuesday — APPROVED** [review]
Looks good, just minor comments

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py (hunk 4 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py]: Addressed

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py (hunk 5 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:44]: Addressed

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py (hunk 6 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py]: Addressed

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py (hunk 7 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py]: Addressed

File: eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py (hunk 8 above)
Comment (sundarvenkata-EBI) [reply eva-accession-release-automation/gather_clustering_counts/gather_clustering_counts_from_mongo.py:392]: Not sure. This is something I observed with taxonomy 8307 where the scientific name is "Ambystoma 'unisexual hybrid'"

**Review by nitin-ebi — APPROVED** [review]

**Pull request merged**